### PR TITLE
maxquant: add checks on experimental design template and documentation

### LINKED ISSUE #392: maxquant regexp problems
```
Template of the MaxQuant parameter file '/gpfs1/work/songalax/galaxy/database/jobs_directory/032/32853/working/mqpar.xml' is created.
Traceback (most recent call last):
  File "/gpfs1/data/galaxy_server/galaxy/shed_tools/toolshed.g2.bx.psu.edu/repos/galaxyp/maxquant/8bac3cc5c5de/maxquant/mqwrapper.py", line 113, in <module>
    description=args['description_parse_rule'])
  File "/gpfs1/data/galaxy_server/galaxy/shed_tools/toolshed.g2.bx.psu.edu/repos/galaxyp/maxquant/8bac3cc5c5de/maxquant/mqparam.py", line 232, in add_fasta_files
    fastaentry = ET.fromstring(fasta)
  File "/gpfs1/data/galaxy_server/galaxy/database/dependencies/_conda/envs/mulled-v1-88194d212d404fa1b66e7626414cd5e03962c42532a28f317b7bc004d3e11dd9/lib/python3.7/xml/etree/ElementTree.py", line 1315, in XML
    parser.feed(text)
xml.etree.ElementTree.ParseError: mismatched tag: line 3, column 38
```

Problem was that the user used `<` and `>` characters in the regular expressions. I guess some validators or replacements are necessary.

### PR DESCRIPTION
**Pull request by dglaetzer**
fixes #392 and #393

### CODE REVIEW

**Comment by dglaetzer** [comment]
@bgruening should this one get a new version number? Something like 1.6.3.4.2 or 1.6.3.4b, to preserve the MaxQuant version in it?

**Comment by bgruening** [comment]
@dglaetzer +galaxy1 would be nice to add, see here: https://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html#tool-versions

This branch should probably be rebased against master. Let me know if you need help here.

File: tools/maxquant/mqparam.py [hunk 1]
```diff
@@ -94,11 +116,17 @@ def _make_exp_design(self, infiles):
                     if i in design:
                         index.append(i)
                     else:
-                        raise Exception("Invalid comlumn index in experimental"
+                        raise Exception("Invalid column index in experimental"
                                         + " design template: {}".format(i))
+                line_no = 0
                 for line in design_file:
```
Comment (bgruening) [inline tools/maxquant/mqparam.py]: use the power of python Luke: http://book.pythontips.com/en/latest/enumerate.html

**Comment by dglaetzer** [comment]
I get a ton of conflicts during rebase. I'll do a new fork, copy my changes in and open a new PR.

**Pull request closed without merging**